src/classes/PMT_Payment_TEST.cls

```diff
@@ -109,6 +109,21 @@ private class PMT_Payment_TEST {
         update opp;
         pmt = getPaymentRecord(opp.Id);
         System.AssertEquals((String)opp.get('CurrencyIsoCode'), (String)pmt.get('CurrencyIsoCode'));
+
+        // Verify that a new Payment cannot be inserted with a different currencyIsoCode
```

Comment:
In order to create clean tests, it is advisable to have one concept tested in one unit test. For this new code, can we create a new test method with name clearly indicating what is being tested and expected?

Reply:
I actually prefer testing one specific set of functionality in the same test. In this case, the test is specifically validating functionality around the CurrencyIsoCode on the Payment object. I think it makes sense to keep these together in a single test.

Reply:
We don't want long test functions that go testing one thing after another. The test is verifying if a new Payment with currency different than a Opprtunity currency will generate an error. The previous code in the same test verifies the Opportunity currency change is updating Payment currency. So if we would create two different methods (preferred, right now we have the "if" control statements to separate the logic) in a class to handle this behaviour, why not create two test methods to test different scenarios?

Moreover, if the first part of the test fails, we'll not know if the second part (related to a different functionality) fails or passes, so we might have two issues but we would not be aware of it until the first one is fixed. If we would not write a long method to verify more than one thing, why would we write a test method to verify more than one thing?

Reply:
Well stated. My past experience with unit tests has generally driven me towards consolidating tests for with related functionality as long as the decision branches are the same. In this case you're absolutely right. The first part of the test does test Opportunity dml while the remaining tests focus on Payment dml. Your second point is just as critical though. A failure in the first part would mask a failure in the second part - at least until the first part is fixed. I think that is acceptable in some situations (such as this one where the tests are extremely small), however there's absolutely value in seeing each individual failure to know that something is not working. If debugging those, it's also satisfying to fix one thing and see that it fixes multiple tests. I'll rework the test class. :)